services/ghibliService.go

```diff
@@ -84,6 +93,54 @@ func filterFilmsById(films [][]string, id string) (models.GhibliFilm, error) {
 	return models.GhibliFilm{}, fmt.Errorf("film with id %s not found in repository", id)
 }
 
+// GetFilm retieves a record from the repository
+func (gs *ghibliService) GetFilmsConcurrently(query url.Values) ([]models.GhibliFilm, error) {
+	typeParam := query.Get("type")
+	maxItemsParam := query.Get("items")
+	maxItemsPerWorkerParam := query.Get("items_per_workers")
+
+	films, err := gs.filsmRepo.ReadCSVFile()
+	if err != nil {
+		log.Printf("Failed to fetch film from repository: %s", err)
+		return []models.GhibliFilm{}, err
+	}
+	var jobs []*workerspool.Job
+	maxItemsPerWorkerNum, _ := strconv.Atoi(maxItemsPerWorkerParam)
+
+	for _, film := range films {
+		jobs = append(jobs, workerspool.NewJob(workerProcess, workerParams{
+			Type:     typeParam,
+			MaxItems: maxItemsPerWorkerParam,
+			Record:   film,
+		}, maxItemsPerWorkerNum))
```

Comment:
why sending max items per worker on each job? You could have passed this on creation